### PR TITLE
Add toc sidebar+dropdown

### LINKED ISSUE #245: Re-add in-document navigation sidebar
**Describe the solution you'd like**
The in-document navigation as shown in the figure below should be added to the react-client.

**Additional context**
This issue requires basic workings on the scrollspy implementation.
The legacy code uses a `ul`-`li` list with some custom styles and `active` classes added by the scrollspy to open sub-menus.

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
The markdown renderer view

### Description
This PR adds the TOC sidebar.

### Steps

- [x] added implementation
- [x] added / updated tests
- [x] added / updated documentation
- [x] extended changelog

### Related Issue(s)
#245

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@testing-library/react": "10.4.3",
     "@testing-library/user-event": "12.0.11",
     "@types/codemirror": "0.0.96",
+    "@types/deep-equal": "^1.0.1",
```
Comment (DerMolly) [inline package.json:10]: Do we want to pin those new dependency directly?

Comment (mrdrogdrog) [reply package.json:10]: done

**Comment by mrdrogdrog** [comment]
@ErikMichelson , @DerMolly 
Please review this PR. I doesn't include the section indicator, because that can only be done after #279

File: package.json (hunk 1 above)
Comment (DerMolly) [reply package.json]: there are quite a few more unpinned?

**Comment by DerMolly** [comment]
I've just run the branch and put the content of the features page in and it show h6 in the toc. Is that intended?
grafik

**Comment by ErikMichelson** [comment]
> Is that intended?

That would be a change then. Currently the features page says (and the implementation is equal):
> You can look at the bottom right section of the view area, there is a ToC button .
Pressing that button will show you a current Table of Contents, and will highlight which section you’re at.
**ToCs support up to three header levels.**

**Comment by mrdrogdrog** [comment]
no, that's something I forgot to add :sweat_smile:

File: package.json (hunk 1 above)
Comment (mrdrogdrog) [reply package.json]: fixed

**Comment by mrdrogdrog** [comment]
@ErikMichelson  done

**Pull request merged**